src/bootstrap/step.rs

```diff
@@ -1209,11 +1209,17 @@ invalid rule dependency graph detected, was a rule added and maybe typo'd?
             if paths.len() == 0 && rule.default {
                 Some((rule, 0))
             } else {
-                paths.iter().position(|path| path.ends_with(rule.path))
+                paths.iter()
+                     .position(|path| path.ends_with(rule.path))
                      .map(|priority| (rule, priority))
             }
         }).collect();
 
+        if rules.is_empty() {
```

Comment:
There are a lot of paths and a lot of them aren't doing anything, so I'm sure this is a good idea...